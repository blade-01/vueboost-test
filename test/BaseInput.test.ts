import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';
import BaseInput from '../src/components/reusables/BaseInput.vue';

describe('Base Input', () => {
  const wrapper = mount(BaseInput, {
    props: {
      label: 'Test',
      type: 'text',
      modelValue: ''
    }
  });
  it('Mounts properly', async () => {
    expect(wrapper).toBeTruthy();
  });
  it('Emits an event when the input is changed', async () => {
    const input = wrapper.find('input');
    await input.setValue('Testing');
    expect(wrapper.emitted('update:modelValue')).toBeTruthy();
  });
  it('Renders the input value', async () => {
    const input = wrapper.find('input');
    await input.setValue('Testing');
    expect(input.element.value).toEqual('Testing');
    expect(input).toMatchSnapshot();
  });
});
