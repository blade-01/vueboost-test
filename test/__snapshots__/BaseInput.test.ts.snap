// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Base Input > Renders the input value 1`] = `
DOMWrapper {
  "isDisabled": [Function],
  "wrapperElement": <input
    class="input-style"
    id="Test"
    type="text"
  />,
}
`;
